plugin/evm/limitorders/memory_database.go

```diff
@@ -39,24 +42,43 @@ const (
 )
 
 type LimitOrder struct {
-	Id                      uint64   `json:"id"`
-	Market                  Market   `json:"market"`
-	PositionType            string   `json:"position_type"`
-	UserAddress             string   `json:"user_address"`
-	BaseAssetQuantity       *big.Int `json:"base_asset_quantity"`
-	FilledBaseAssetQuantity *big.Int `json:"filled_base_asset_quantity"`
-	Salt                    *big.Int `json:"salt"`
-	Price                   *big.Int `json:"price"`
-	Status                  Status   `json:"status"`
-	Signature               []byte   `json:"signature"`
-	RawOrder                interface{}
-	BlockNumber             *big.Int `json:"block_number"` // block number order was placed on
+	Id                          uint64                   `json:"id"`
+	Market                      Market                   `json:"market"`
+	PositionType                string                   `json:"position_type"`
+	UserAddress                 string                   `json:"user_address"`
+	BaseAssetQuantity           *big.Int                 `json:"base_asset_quantity"`
+	FilledBaseAssetQuantity     *big.Int                 `json:"filled_base_asset_quantity"`
+	Salt                        *big.Int                 `json:"salt"`
+	Price                       *big.Int                 `json:"price"`
+	Status                      Status                   `json:"status"`
+	Signature                   []byte                   `json:"signature"`
+	BlockNumber                 *big.Int                 `json:"block_number"`                    // block number order was placed on
+	InProgressBaseAssetQuantity map[common.Hash]*big.Int `json:"in_progress_base_asset_quantity"` // block hash => quantity
+	RawOrder                    interface{}              `json:"-"`
 }
 
 func (order LimitOrder) GetUnFilledBaseAssetQuantity() *big.Int {
 	return big.NewInt(0).Sub(order.BaseAssetQuantity, order.FilledBaseAssetQuantity)
 }
 
+// GetAvailableUnFilledBaseAssetQuantity returns (total quantity - filled quantity - in progress quantity) - signed
+func (order LimitOrder) GetAvailableUnFilledBaseAssetQuantity() *big.Int {
+	if order.PositionType == "long" {
+		return big.NewInt(0).Sub(order.GetUnFilledBaseAssetQuantity(), order.GetInProgressBaseAssetQuantity())
+	} else {
+		return big.NewInt(0).Add(order.GetUnFilledBaseAssetQuantity(), order.GetInProgressBaseAssetQuantity())
```

Comment:
while this is correct, it's a bit inconsistent code-readability-wise. It's better to set `-fillAmount` for the short order in the orderQuantityMap and then always subtract here. It will become same as `GetUnFilledBaseAssetQuantity`